src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,594 @@
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidAlgorithmParameterException. Does this "
+                + "algorithm require KDFParameters?", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm from
+     * the specified security provider.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if the specified provider does not support the specified KDF
+     *     algorithm
+     * @throws NoSuchProviderException
+     *     if the specified provider is not registered in the security provider
+     *     list
+     * @throws NullPointerException
+     *     if the algorithm or provider is {@code null}
+     */
+    public static KDF getInstance(String algorithm, String provider)
+        throws NoSuchAlgorithmException, NoSuchProviderException {
+        Objects.requireNonNull(provider, "provider may not be null");
+        try {
+            return getInstance(algorithm, null, provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidAlgorithmParameterException. Does this "
+                + "algorithm require KDFParameters?", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm from
+     * the specified security provider.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if the specified provider does not support the specified KDF
+     *     algorithm
+     * @throws NullPointerException
+     *     if the algorithm or provider is {@code null}
+     */
+    public static KDF getInstance(String algorithm, Provider provider)
+        throws NoSuchAlgorithmException {
+        Objects.requireNonNull(provider, "provider may not be null");
+        try {
+            return getInstance(algorithm, null, provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidAlgorithmParameterException. Does this "
+                + "algorithm require KDFParameters?", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm and
+     * is initialized with the specified parameters.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param kdfParameters
+     *     the {@code KDFParameters} used to configure this KDF's algorithm or
```

Comment:
Just say "configure this KDF". Not sure what "configure this KDF's algorithm" means exactly.